docs/INSTALL.md

```diff
@@ -1,9 +1,69 @@
 # Building
 
+- [Getting sources](#getting-sources)
 - [Desktop](#desktop-app)
 - [Android](#android-app)
 - [iOS](#ios-app)
 
+## Getting sources
+
+First of all get the source code. The full Organic Maps sources repository is ~8.5Gb in size, however its possible to reduce the size e.g. if full commit history is not required.
+
+For _Windows 10_ enable [symlinks][git-symlinks] support in git first:
+
+[git-symlinks]: https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresymlinks
+
+```bash
+git config --global core.symlinks true
+```
+
+If you plan to contribute and propose pull requests then fork the repo https://github.com/organicmaps/organicmaps first.
+
+Clone the main repository (~5.5Gb):
+
+```bash
+git clone https://github.com/organicmaps/organicmaps.git
+```
+or your fork of it:
+```bash
+git clone https://github.com/<your-github-name>/organicmaps.git
+```
+
+Add a `--filter=blob:limit=128k` option to above command to make a _partial clone_ of just ~1.2Gb, i.e. blob files over 128k in size will be excluded form the history and downloaded on-demand - should be suitable for a generic development use.
```

Comment:
Move to contribute block. But is it really necessary? If you want to contribute, then cloning everything is the most reliable way.

s/form/from/

Reply:
Its not necessary of course, its just a way to lower barriers for people with low-spec machines and limited/slow Internet.

E.g. my laptop's SSD is 120Gb, devoting 30+Gb to be able to contribute to OM is a lot.
Cloning the full repo took me an hour almost and I was lucky to have an unlimited wifi for that - usually I'm on a limited plan.

Though I agree those options could be moved somewhere in footnotes or a special subsection perhaps.

Reply:
Why 30+ GB? My full checked-out repo takes ~9GB (7GB .git folder). Every Android arch debug build takes an additional 4.5Gb. Building a desktop-only app takes also a few GBs, depending on the configuration.

Reply:
30+Gb is a minimum total that also includes android studio, sdk, ndk, etc.

On my system it looks like this:
>		1.6 android studio
>		8.5 full repo
>		9 sdk, ndk, emulator image
>		1.5 ~/.android/avd (emulated storage etc?)
>		= 20.5Gb
>		
>		+1.5 ~./gradle/caches
> 		+4.5*3=13.5 build files for 3 arches
>		= 35.5Gb total

There I have a small emulator image without Google services and a quite old low-mem emulated device (Nexus S) and just a 256Mb emulated SD. With a more conventional option it'll be a couple Gb more.

So actually if we expect someone to just follow default instructions and build for all 3 arches (which is by default) then it comes close to 40Gb...

Reply:
Interesting. We probably need to document a "low space" options, e.g. installing only JDK without Android Studio (or reusing a system one), building release instead of debug to save some space, mounting a compressed NDK image :)

But let's start focusing on systems with enough free space first and make it an easy start for these environments.